pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -120,18 +119,21 @@ func (c *controller) reconcilePermissionClaims(ctx context.Context, apiBinding *
 			}
 		}
 		if !found {
-			//TODO: add openAPI check such that this could not be incorrect.
-			if acceptedPC.IdentityHash != "" {
-				if grsToBoundResource[acceptedPC.GroupResource].Schema.IdentityHash != acceptedPC.IdentityHash {
-					identityMismatch = true
-					invalidClaims = append(invalidClaims, acceptedPC)
-					continue
-				}
+			ok, err := permissionclaims.ValidateClaim(acceptedPC.Group, acceptedPC.Resource, acceptedPC.IdentityHash, grsToBoundResource)
+			if !ok {
+				invalidClaims = append(invalidClaims, acceptedPC)
+				identityMismatch = true
+				klog.V(3).Infof("invalid claim %v - reason %v", acceptedPC, err)
```

Comment:
klog.V(3).InfoS

Reply:
we should only add structured logs from now on that we have 1.24 kube.

Reply:
Is there a guide on structured logging? can I use fmt.Sprintf in those log statements.

Reply:
fmt.Sprintf as individual call is forbidden in logs, even before structured logs. 

Just grep for InfoS to find good example. Kube has many.

Reply:
Attempted to add all of these